til-domain/src/main/java/com/til/domain/problem/repository/AdminProblemRepositoryCustomImpl.java

```diff
@@ -0,0 +1,202 @@
+import static com.til.domain.problem.repository.ProblemQueryCondition.*;
+import static com.til.utils.data.ListUtil.isNullOrEmpty;
+import static com.til.utils.data.StringUtil.hasText;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+import com.querydsl.core.group.GroupBy;
+import com.querydsl.core.types.Order;
+import com.querydsl.core.types.OrderSpecifier;
+import com.querydsl.core.types.Projections;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.core.types.dsl.PathBuilder;
+import com.querydsl.jpa.JPAExpressions;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import com.querydsl.jpa.impl.JPAUpdateClause;
+import com.til.domain.common.exception.BaseException;
+import com.til.domain.problem.dto.*;
+import com.til.domain.problem.enums.ProblemErrorCode;
+import com.til.domain.problem.enums.ProblemSortCriteria;
+import com.til.domain.problem.model.QProblem;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class AdminProblemRepositoryCustomImpl implements AdminProblemRepositoryCustom {
+
+    private final JPAQueryFactory queryFactory;
+
+    @Override
+    @Transactional
+    public void updateProblem(AdminUpdateProblemDto adminUpdateProblemDto) {
+        QProblem problem = QProblem.problem;
+
+        JPAUpdateClause updateClause = queryFactory.update(problem)
+            .where(problem.id.eq(adminUpdateProblemDto.id()));
+
+        boolean hasUpdate = false;
+
+        if (adminUpdateProblemDto.title() != null ||
+            adminUpdateProblemDto.question() != null ||
+            adminUpdateProblemDto.solution() != null ||
+            adminUpdateProblemDto.grading() != null ||
+            adminUpdateProblemDto.level() != null) {
+
+            if (adminUpdateProblemDto.title() != null) {
+                updateClause.set(problem.title, adminUpdateProblemDto.title());
+                hasUpdate = true;
+            }
+            if (adminUpdateProblemDto.question() != null) {
+                updateClause.set(problem.question, adminUpdateProblemDto.question());
+                hasUpdate = true;
+            }
+            if (adminUpdateProblemDto.solution() != null) {
+                updateClause.set(problem.solution, adminUpdateProblemDto.solution());
+                hasUpdate = true;
+            }
+            if (adminUpdateProblemDto.grading() != null) {
+                updateClause.set(problem.grading, adminUpdateProblemDto.grading());
+                hasUpdate = true;
+            }
+            if (adminUpdateProblemDto.level() != null) {
+                updateClause.set(problem.level, adminUpdateProblemDto.level());
+                hasUpdate = true;
+            }
+        }
```

Comment:
P4: 제일 처음 if으로 감싸지 않고 내부 if문 만 활용해도 동작이 가능할 것 같습니다!
```suggestion
        if (adminUpdateProblemDto.title() != null) {
            updateClause.set(problem.title, adminUpdateProblemDto.title());
            hasUpdate = true;
        }
        if (adminUpdateProblemDto.question() != null) {
            updateClause.set(problem.question, adminUpdateProblemDto.question());
            hasUpdate = true;
        }
        if (adminUpdateProblemDto.solution() != null) {
            updateClause.set(problem.solution, adminUpdateProblemDto.solution());
            hasUpdate = true;
        }
        if (adminUpdateProblemDto.grading() != null) {
            updateClause.set(problem.grading, adminUpdateProblemDto.grading());
            hasUpdate = true;
        }
        if (adminUpdateProblemDto.level() != null) {
            updateClause.set(problem.level, adminUpdateProblemDto.level());
            hasUpdate = true;
        }
```